formatter/src/main/java/org/togetherjava/formatter/CodeSectionFormatter.java

```diff
@@ -0,0 +1,396 @@
+     * @author illuminator3
+     */
+    private static void purgeWhitespaces(Queue<Token> queue) {
+        queue.removeIf(t -> t.type() == TokenType.WHITESPACE);
+    }
+
+    /**
+     * Starts the formatting process
+     *
+     * @author illuminator3
+     */
+    void format() {
+        Token next;
+
+        while (!queue.isEmpty() && (next = queue.remove()) != null) {
+            updateIndentation(next.type());
+            applyIndentation();
+            consume(next);
+        }
+    }
+
+    /**
+     * Consumes the next token
+     *
+     * @param token token to consume
+     * @author illuminator3
+     */
+    private void consume(Token token) {
+        TokenType type = token.type();
+
+        if (checkGeneric(type)) {
+            result.append(token.content());
+
+            return;
+        }
+
+        if (genericDepth != 0) {
+            handleGeneric(token);
+
+            return;
+        }
+
+        checkFor(type);
+        updateEnhancedLevel(type);
+        put(token);
+    }
+
+    /**
+     * Puts the next token into the result
+     *
+     * @param token token to put
+     * @author illuminator3
+     */
+    private void put(Token token) {
+        TokenType type = token.type();
+
+        if (isOperator(token)) { // apply space before an operator (like +, - and *)
+            result.append(' ');
+        }
+
+        result.append(token.content());
+
+        if (shouldPutSpaceAfter(token)) {
+            result.append(' ');
+        } else if (shouldPutNewLineAfter(type)) { // put a new line after { and ; and } and comments
+                                                  // and annotations
+            appendNewLine();
+
+            if (type == TokenType.SEMICOLON) {
+                forLevel--;
+            }
+
+            applyIndentation = true;
+        }
+    }
+
+    /**
+     * Checks if a space should be put after a given token
+     *
+     * @param token token to check
+     * @return wether a space should be put after that token
+     * @author illuminator3
+     */
+    private boolean shouldPutSpaceAfter(Token token) {
+        TokenType type = token.type();
+
+        return isKeyword(token) || isOperator(token) || isParenthesisRule(token)
+                || type == TokenType.CLOSE_BRACKETS // put a space after e.g. 'try' or 'else' and
+                                                    // ] or if it's an operator
+                || type == TokenType.COMMA // for e.g. multiarg method calls or method parameters
+                || (type == TokenType.IDENTIFIER && !queue.isEmpty()
+                        && queue.peek().type() == TokenType.IDENTIFIER); // for double identifier
+                                                                         // in e.g. method
+                                                                         // declaration or enhanced
+                                                                         // for loops
+    }
```

Comment:
Also, I dislike inline comments, especially when they do this. :shrug:

Reply:
Where else am I supposed to put those comments? They explain what each part of the conditional does.